include/drivers/gpio.h

```diff
@@ -542,6 +564,64 @@ static inline int z_impl_gpio_disable_callback(struct device *port,
  * @endcond
  */
 
+/**
+ * @brief Configure pin interrupt.
+ *
+ * @note This function can also be used to configure interrupts on pins
+ *       not controlled directly by the GPIO module. That is, pins which are
+ *       routed to other modules such as I2C, SPI, UART.
+ *
+ * @param port Pointer to device structure for the driver instance.
+ * @param pin Pin number.
+ * @param flags Interrupt configuration flags as defined by GPIO_INT_*.
+ *
+ * @retval 0 If successful.
+ * @retval -ENOTSUP If any of the configuration options is not supported.
+ * @retval -EINVAL  Invalid argument.
+ * @retval -EBUSY   Interrupt line required to configure pin interrupt is
+ *                  already in use.
+ */
+__syscall int gpio_pin_interrupt_configure(struct device *port,
+		unsigned int pin, unsigned int flags);
+
+static inline int z_impl_gpio_pin_interrupt_configure(struct device *port,
+		unsigned int pin, unsigned int flags)
+{
+	const struct gpio_driver_api *api =
+		(const struct gpio_driver_api *)port->driver_api;
+	struct gpio_driver_data *const data =
+			(struct gpio_driver_data *const)port->driver_data;
+
+	__ASSERT(pin < GPIO_MAX_PINS_PER_PORT, "Invalid pin number");
+
+	__ASSERT(((flags & GPIO_INT_ENABLE) == 0) ||
+		 ((flags & (GPIO_INT_LOW_0 | GPIO_INT_HIGH_1)) != 0),
+		 "At least one of GPIO_INT_LOW_0, GPIO_INT_HIGH_1 has to be "
+		 "enabled.");
+
+	flags = flags & GPIO_INT_ALL_MASK;
+	if ((flags & GPIO_INT_LEVELS_LOGICAL) && (data->invert & BIT(pin))) {
+		/* Invert signal bits */
+		flags = ~(flags & (GPIO_INT_LOW_0 | GPIO_INT_HIGH_1)) |
+			(flags & ~(GPIO_INT_LOW_0 | GPIO_INT_HIGH_1));
+	}
+
+	/* Clear GPIO_INT_LEVELS_LOGICAL */
+	flags &= ~GPIO_INT_LEVELS_LOGICAL;
+
+	if ((flags == GPIO_INT_MODE_DISABLED) ||
+	    (flags == GPIO_INT_LEVEL_LOW) ||
```

Comment:
Shouldn't this be `GPIO_INT_MODE_LEVEL_LOW`?

Reply:
Yes, good catch.

Reply:
Fixed.